gatherers/manifest.js

```diff
@@ -54,34 +38,87 @@ class Manifest extends Gather {
+      resolve,
+      reject
+    };
+  }
+
+  /**
+   * Creates a listener for manifests in network requests, a load timeout, and
+   * a promise they resolve when complete.
+   * @private
+   */
+  static _resolveOnManifest() {
+    const manifestDeferred = Manifest._createDeferred();
+
+    setTimeout(_ => manifestDeferred.resolve(null), MANIFEST_LOAD_TIMEOUT);
+
+    const manifestListener = function(request) {
+      const resourceType = request.resourceType();
+      if (resourceType && resourceType.name() === 'manifest') {
+        manifestDeferred.resolve(request);
+      }
+    };
+
+    return {
+      listener: manifestListener,
+      promise: manifestDeferred.promise
+    };
+  }
+
   static gather(options) {
     const driver = options.driver;
-    /**
-     * This re-fetches the manifest separately, which could
-     * potentially lead to a different asset. Using the original manifest
-     * resource is tracked in issue #83
-     */
-    return driver.querySelector('head link[rel="manifest"]')
-      .then(node => {
-        if (!node) {
-          return this._errorManifest('No <link rel="manifest"> found in DOM.');
+
+    const manifestLoad = Manifest._resolveOnManifest();
+
+    return driver.beginNetworkCollect(manifestLoad.listener)
+      // Trigger a fetch of the manifest by Chrome.
+      .then(_ => driver.sendCommand('Page.requestAppBanner'))
+
+      // Wait on grabbing the finished manifest network request (or timing out).
+      .then(_ => manifestLoad.promise)
+
+      .then(manifestRequest => {
+        if (!manifestRequest) {
+          return Manifest._errorManifest('Timed out waiting for manifest to load.');
+        }
+        if (manifestRequest.statusCode >= 400) {
+          const reason = `${manifestRequest.statusCode}: ${manifestRequest.statusText}`;
+          return Manifest._errorManifest(
+              `Unable to fetch manifest at '${manifestRequest.url}' (${reason}).`);
         }
 
-        return node.getAttribute('href').then(manifestURL => {
-          if (!manifestURL) {
-            return this._errorManifest('No href found on <link rel="manifest">.');
-          }
-
-          return Manifest._loadFromURL(options, manifestURL)
-            .then(manifestContent => {
-              return {
-                manifest: manifestParser(manifestContent)
-              };
-            })
-            .catch(reason => {
-              return this._errorManifest(`Unable to fetch manifest at ${manifestURL}: ${reason}.`);
-            });
-        });
+        // On successful request, grab the manifest request body for parsing.
+        return driver.sendCommand('Network.getResponseBody', {
+          requestId: manifestRequest.requestId
+        })
+          .then(response => {
+            return {
+              manifest: manifestParser(response.body)
+            };
+          })
+          .catch(_ => {
+            return Manifest._errorManifest('Network recorder unable to find request for ' +
```

Comment:
see https://github.com/GoogleChrome/lighthouse/pull/137#discussion_r58642451
this design won't work for how the manifest needs to be captured